nebula_scripts/Build_cabal_and_project.py

```diff
@@ -0,0 +1,105 @@
+# This script is used to modify the cabal file by change or add a build depends and ghc-option
+# This script also builds a cabal.project  
+# Three arguments needed:
+# first argument is the python file
+# second argument is the directory that contain all the package that contain rules
+# third argument is the location of g2 in one's computer so we can 
+import os 
+import sys
+import re
+import tempfile
+
+def changing_cabal(directory):
+    found_ghc = False
+    found_build = False
+    for root, dirs, files in os.walk(directory):
+        for file_name in files:
+            # process for replace cabal file 
+            file_path = os.path.join(root,file_name)
+            #print("The file_path is " + file_path)
+            if file_path.endswith('.cabal'):
+                # make a file directly in the directory where the cabal file is located: root?
+                temp_path_info =  tempfile.mkstemp(dir=os.path.dirname(file_path))
+                temp_path = temp_path_info[1]
+                # writing information of the cabal file into a temp file and then replace the build depends
+                try:
+                    with open(file_path,'r') as file, open(temp_path,'a') as temp_file:
+                        for line in file:
+                            search_build_depends = re.search("Build-Depends:",line,re.IGNORECASE)
```

Comment:
A complication here is that a cabal file might be building MULTIPLE projects, each with its own build-depends/ghc-options.  Might need re.findall (and then a loop) rather than re.search?